### PR TITLE
Include Docker Compose 

### PR DESCRIPTION
**Pull request by akath20**
Working on dockerizing the application with compose. Currently facing ~90% cpu from the application imagine when using docker-compose, need to check in with @7imbrook how to resolve before merging. #57

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: "2"
+services:
+    mongo:
```
Comment (7imbrook) [inline docker-compose.yml]: will need 
```
deploy:
    constraints:
        - node.labels.db=true
```
if planning on running our own

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        image: mongo
+        volumes:
+            - ./data:/data/db
+        ports:
+            - "27017:27017"
+    web:
+        build:
+            context: .
```
Comment (7imbrook) [inline docker-compose.yml]: Additionally this is good for dev, but we'll also need an image only one for prod deploys.

File: docker-compose.yml (hunk 1 above)
Comment (7imbrook) [reply docker-compose.yml]: its something like OneRepos config

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            dockerfile: Dockerfile.dev
+        ports:
+            - "8080:8080"
+        links:
+            - mongo
+        volumes:
+            - .:/usr/src/app
```
Comment (7imbrook) [inline docker-compose.yml]: Same as above

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+version: "2"
```
Comment (7imbrook) [inline docker-compose.yml]: Update to version 3.1

File: docker-compose.yml (hunk 1 above)
Comment (7imbrook) [reply docker-compose.yml]: cause you'll also have to add
```
volumes:
    volumeName:
```

File: docker-compose.yml (hunk 3 above)
Comment (7imbrook) [reply docker-compose.yml]: I understand that this is a dev config

File: Dockerfile.dev [hunk 5]
```diff
@@ -0,0 +1,11 @@
+FROM node:6.10.1
+
+WORKDIR /usr/src/app
+
+ADD start.sh .
+
+RUN chmod +x ./start.sh
```
Comment (7imbrook) [inline Dockerfile.dev]: Permissions are preserved and should be maintained in source. So change this outside the build.

**Review by 7imbrook — DISMISSED** [review]
Left some comments, mainly to move it to a prod config from a dev config.

**Comment by akath20** [comment]
Can't believe it, but everything is working with this push. The first workaround that needs to be addressed though, which I would need help by likely both @7imbrook and @khanny17 on figuring out where I'm going wrong is the config.js is not being added to the container correctly. I believe this is because I am working with the local file as context, and not dist/. It's trying to access config.js, but it's not actually there/not being updated correctly. The second issue is high CPU usage...

**Comment by akath20** [comment]
I fixed the above issue of the config file. This is fully working, I just need to investigate the CPU issue...

**Comment by akath20** [comment]
@7imbrook have you ever encountered com.docker.hyperkit acting up? 
https://github.com/docker/for-mac/issues/1094
https://forums.docker.com/t/com-docker-hyperkit-up-cpu-to-340/13057/68
I'm not sure if this might be fixed in the beta channel you're on? It's working, I'm just concerned as I mentioned today it might cause issues on the server, even though we can mask the issue. It seems like a bug in Docker beyond our control at the moment...

**Comment by khanny17** [comment]
Did you confirm that the high cpu issue is due to docker and not due to gulp devrun?

**Comment by akath20** [comment]
This looks good to go, and with this fix should fix the production issue. Once this is merged to master, I can redeploy and test.

**Review by khanny17 — APPROVED** [review]

**Pull request merged**